app/src/components/account/UserMenu.vue

```diff
@@ -49,22 +53,42 @@
           <div class="address-item">
             <img
               class="network-icon"
-              :src="address.icon"
+              :src="address.icon || `/img/networks/${address.networkId}.png`"
               alt="little circle with network logo"
             />
             <div>
-              <span class="address-network">{{
-                getAddressNetwork(address).title
-              }}</span>
+              <div class="address-name-role">
+                <p class="address-name">
+                  {{ getSessionName(address) || `Unknown` }}
+                </p>
+                <p
+                  v-if="
+                    getAddressNetwork(address).network_type === `polkadot` &&
```

Comment:
why do we have to get the network for the address if we already have `network_type`? 

this is kind of a mess... can we enrich the polkadot addresses in the keystore instead of in the user menu? the user menu should really just display data.

Reply:
well, actually this check with the ` network_type` here is not even necessary, since all other networks have `none` as their addressRole, so we can actually remove this one check

Reply:
but enrich how? AddressRole is dynamic. I tried this approach before but it just doesn't work

Reply:
but yeah, maybe it doesn't make that much sense having it in there right? Because it is anyway a typical function receiving some parameter and returning something.

I really do see your point here, Jordan, but in order for it to make sense I think we would need to add all addresses also to the `session` module in store, and then enrich them there.

Reply:
Done